marklogic-data-hub/src/main/java/com/marklogic/hub/hubcentral/HubCentralManager.java

```diff
@@ -31,33 +33,93 @@
 public class HubCentralManager extends LoggingObject {
 
     public void writeProjectArtifactsAsZip(HubClient hubClient, OutputStream outputStream) {
-        ArrayNode artifacts = (ArrayNode) ArtifactService.on(hubClient.getStagingClient()).getArtifactsWithProjectPaths();
-
         final ZipOutputStream zipOutputStream = new ZipOutputStream(outputStream);
-        final ObjectWriter prettyWriter = buildPrettyWriter();
+        try {
+            writeArtifactsIntoZipOutputStream(hubClient, zipOutputStream);
+        } finally {
+           IOUtils.closeQuietly(zipOutputStream);
+        }
+    }
 
+    public void writeProjectAsZip(HubClient hubClient, OutputStream outputStream){
```

Comment:
I think this needs less knowledge of what's actually in the project, as this is duplicating knowledge that initProject has in HubProject. How about the following approach instead:

1. Create a temp directory using java.io.tmpdir
1. Use HubProject to initialize a project in that temp dir
1. Get all of the project update files and write them over top of the initialized project dir
1. Zip up the entire project directory 

Now, if we add a new project file in initProject in HubProject, we don't have to remember to account for it here too.

Reply:
I have used hubProject.exportProject() for step 4.